br/tests/rawkv/main.go

```diff
@@ -222,82 +251,100 @@ func CheckBackupTS(apiVersion kvrpcpb.APIVersion, tso uint64, backupOutput []byt
+	backupOutput, err := tester.Backup(ctx, tester.apiVersion, safeInterval, start, end)
 	if err != nil {
 		log.Panic("Backup fail", zap.Error(err), zap.ByteString("output", backupOutput))
 	}
 	log.Info("backup finish:", zap.ByteString("output", backupOutput))
-	CheckBackupTS(apiVersion, tso, backupOutput, safeInterval)
+	CheckBackupTS(tester.apiVersion, tso, backupOutput, safeInterval)
 
 	err = tester.CleanData(ctx, prefix)
 	if err != nil {
 		log.Panic("Clean data fail", zap.Error(err))
 	}
-	restoreOutput, err := tester.Restore(ctx)
+	restoreOutput, err := tester.Restore(ctx, start, end)
 	if err != nil {
 		log.Panic("Restore fail", zap.Error(err), zap.ByteString("restore output", restoreOutput))
 	}
 	log.Info("Restore finish:", zap.ByteString("output", restoreOutput))
-	restoreChecksum, err := tester.Checksum(ctx)
+	restoreChecksum, err := tester.Checksum(ctx, start, end)
 	if err != nil {
 		log.Panic("Checksum fail", zap.Error(err))
 	}
 	if oriChecksum != restoreChecksum {
 		log.Panic("Checksum mismatch", zap.Reflect("src", oriChecksum), zap.Reflect("dst", restoreChecksum))
 	}
 	log.Info("Checksum pass")
+}
+
+func runTestWithFailPoint(failpoint string) {
+	apiVersion := kvrpcpb.APIVersion_V1TTL
+	if *apiVersionInt == 2 {
+		apiVersion = kvrpcpb.APIVersion_V2
+	}
+	ctx := context.TODO()
+
+	fmt.Println("test api version", apiVersion)
+
+	tester, err := NewRawKVBRTester(ctx, *pdAddr, *br, *brStorage, apiVersion)
+	if err != nil {
+		log.Panic("New Tester Fail", zap.Error(err))
+	}
+	defer tester.Close()
+
+	err = tester.InjectFailpoint(failpoint)
+	if err != nil {
+		log.Panic("Inject failpoint fail", zap.Error(err), zap.String("failpoint", failpoint))
+	}
 
-	if apiVersion == kvrpcpb.APIVersion_V1TTL {
+	prefix := []byte("index")
+	q1Key, _ := GenerateTestData(*keyCnt/4, prefix)
+	q3Key, _ := GenerateTestData(3**keyCnt/4, prefix)
+	backupRanges := []kvrpcpb.KeyRange{
+		{StartKey: []byte{}, EndKey: []byte{}},
+		{StartKey: q1Key, EndKey: q3Key},
+	}
+	for _, keyRange := range backupRanges {
 		if err := tester.ClearStorage(); err != nil {
 			log.Panic("ClearStorage fail", zap.Error(err))
 		}
-		backupOutput, err := tester.Backup(ctx, kvrpcpb.APIVersion_V2, safeInterval)
+		err = tester.CleanData(ctx, prefix)
+		if err != nil {
+			log.Panic("Clean data fail", zap.Error(err))
+		}
+		err = tester.PreloadData(ctx, *keyCnt, *thread, prefix)
 		if err != nil {
-			log.Panic("Backup fail", zap.Error(err), zap.ByteString("backup output", backupOutput))
+			log.Panic("Preload data fail", zap.Error(err))
+		}
+		runBackupAndRestore(ctx, tester, prefix, keyRange.StartKey, keyRange.EndKey)
+
+		if apiVersion == kvrpcpb.APIVersion_V1TTL {
+			if err := tester.ClearStorage(); err != nil {
+				log.Panic("ClearStorage fail", zap.Error(err))
+			}
+			safeInterval := int64(120) // 2m
+			backupOutput, err := tester.Backup(ctx, kvrpcpb.APIVersion_V2, safeInterval, keyRange.StartKey, keyRange.EndKey)
```

Comment:
Suggest to also verify the backup result by restore + checksum.

Reply:
It's a bit problem here, we only deploy one cluster in integration test. May need deploy apiv1 & apiv2 clusters to verify the conversion result.

Reply:
Got it. We may consider to deploy two clusters for this test case later.

Reply:
Ok，I will improve it later.